website/docs/standalone-gallery.md

```diff
@@ -0,0 +1,221 @@
+slug: /apis/standalone-gallery
+---
+
+`Standalone Gallery` renders Pager which supports thousands of images thanks to virtualization. Each page renders `ImageTransformer` component which gives ability to pinch-to-zoom, double tap to zoom, also you can run custom callbacks and worklets on on tab, double tap, pan.
+
+> TODO: add preview here
+
+## Usage
+
+```tsx
+import React from 'react';
+import {
+  StandaloneGallery,
+  GalleryItemType,
+  StandaloneGalleryHandler,
+} from 'react-native-gallery-toolkit';
+
+const images: GalleryItemType[] = [
+  {
+    id: '1',
+    width: 300,
+    height: 300,
+    uri: 'https://placekitten.com/300/300',
+  },
+  {
+    id: '2',
+    width: 400,
+    height: 200,
+    uri: 'https://placekitten.com/400/200',
+  },
+];
+
+export default function App() {
+  return <StandaloneGallery items={images} />;
+}
+```
+
+## Base Props
+
+### `items`
+
+The array of items to render. But can also accept `Map`, `Set`, or `Object` with keys. If the type is not array, then `getTotalCount` and `getItem` should be defined too.
+
+type | default | required
+------ | ------ | ------
+`Array<GalleryItemType>` | `undefined` | Yes
+
+### `width`
+
+Viewport width, usually it will be the window width.
+
+type | default | required
+------ | ------ | ------
+`number` | `Dimensions.get('window').width` | NO
+
+### `height`
+
+Viewport height, usually it will be the window height.
+
+type | default | required
+------ | ------ | ------
+`number` | `Dimensions.get('window').height` | NO
+
+### `numToRender`
+
+How many pages should be rendered at the same time.
+
+type | default | required
+------ | ------ | ------
+`number` | `2` | NO
+
+### `gutterWidth`
+
+The width of the gutter between pages.
+
+type | default | required
+------ | ------ | ------
+`number` | `0` | NO
+
+### `initialIndex`
+
+The initial page index.
+
+type | default | required
+------ | ------ | ------
+`number` | `0` | NO
+
+### `keyExtractor`
+
+Callback which extract the key of the page. Receives current item of the provided items as well as current index.
+
+type | default | required
+------ | ------ | ------
+`(item: T, index: number) => string` | Uses index as a key by default | NO
+
+## Advance Props
+
+### `getTotalCount`
+
+If the type of `items` is not an array, then this method should be defined to provide the total count of items
+
+type | default | required
+------ | ------ | ------
+`(data: T) => number` | `undefined` |  Required when items is not an array
+
+### `getItem`
+
+If the type of `items` is not an array, then this method should be defined to provide the current item based on the index. Can return either the `item` or `undefined`.
+
+type | default | required
+------ | ------ | ------
+`(data: T, index: number) => ItemT or undefined` | `undefined` | Required when items is not an array
+
+### `renderImage`
+
+Callback that can be used to render custom image component. As an example, it can be used to render custom loading/error states.
```

Comment:
```suggestion
Callback that can be used to render custom image component. As an example, it can be used to render custom loading/error states or if you want to use custom Image component (FastImage, for example).

By default it uses Image component provided by React Native.
```